zerver/tests/test_auth_backends.py

```diff
@@ -1066,7 +1066,11 @@ def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomai
             self.assert_not_in_success_response(['id_password'], result)
             self.assert_in_success_response(['id_full_name'], result)
             # Verify the name field gets correctly pre-populated:
-            self.assert_in_success_response([expected_final_name], result)
+            if (self.BACKEND_CLASS != AppleAuthBackend):
```

Comment:
Makes sense, though unfortunate we have to do that :cry: What we can do to avoid polluting this helper function with this apple-specific logic, I think we can do a different trick and override `stage_two_of_registration` in the apple backend to simply call `super().stage_two_of_registration(...)` with `expected_final_name=''`? Same effect, though more hacky - but we manage to keep this helper cleaner at least, so might be worth it

Reply:
I'm not sure if I understood the right way about override, you'e saying that we'll just make the content of the overriden function this?
``` 
super().stage_two_of_registration(......,expected_final_name='')
```
If we do that it'd fail the test because we cannot have full name here to be empty string I guess.
```
# Click confirm registration button.
            result = self.client_post(
                '/accounts/register/',
                {'full_name': expected_final_name,
                 'key': confirmation_key,
                 'terms': True})
```
Now I see that that change is only for one test `test_auth_registration_with_no_name_sent_from_apple`. so, I'll copy the contents of `stage_two_of registration` to only have relevant parts.